chain/network/src/routing/routing_table_actor.rs

```diff
@@ -306,63 +306,58 @@ impl RoutingTableActor {
         prune_edges_not_reachable_for: Duration,
     ) -> Vec<Edge> {
         let now = Instant::now();
-        let mut oldest_time = now;
+        // Save nodes on disk and remove from memory only if elapsed time from oldest peer
+        // is greater than `SAVE_PEERS_MAX_TIME`
+        let do_pruning = force_pruning
+            || (self.peer_last_time_reachable.values())
+                .any(|&last_time| now.saturating_duration_since(last_time) >= SAVE_PEERS_MAX_TIME);
```

Comment:
@matklad Btw, this code is actually faster. Most of the time this condition is false, so in most common case, we will avoid doing extra memory allocations.